indexing-service/src/main/java/io/druid/indexing/common/actions/SegmentTransactionalInsertAction.java

```diff
@@ -99,14 +99,24 @@ public DataSourceMetadata getEndMetadata()
   @Override
   public SegmentPublishResult perform(Task task, TaskActionToolbox toolbox) throws IOException
   {
-    toolbox.verifyTaskLocks(task, segments);
+    for (DataSegment segment : segments) {
+      if (!toolbox.getTaskLockbox().upgrade(task, segment.getInterval()).isOk()) {
+        return SegmentPublishResult.fail();
```

Comment:
I think you could make SegmentPublishResult.fail() a static object instead of a function that returns a new one